### PR TITLE
Added example showing transferability within and across models using pure tf.

### PR DESCRIPTION
**Pull request by abhibhav14**
The example uses the classes from the pure tf implementation (#151).

### CODE REVIEW

**Comment by abhibhav14** [comment]
I signed it!

File: examples/ex_mnist_transfer_puretf.py [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+    adv_x = fgsm.generate(x, **fgsm_params)
+    preds_adv = model.fprop(adv_x)
+
+    # Evaluate the accuracy of the MNIST model on adversarial examples
+    eval_par = {'batch_size': batch_size}
+    acc = model_eval(sess, x, y, preds_adv, X_test, Y_test, args=eval_par)
+    print('Test accuracy on adversarial examples: %0.4f\n' % acc)
+    
+    # Print confusion Matrix
+    confusion(sess, x, y, preds_adv, X_test, Y_test, batch_size=batch_size)
```
Comment (npapernot) [inline examples/ex_mnist_transfer_puretf.py]: Right now, when running the script it is somewhat hard to understand to which pair of models a confusion matrix refers to. Could you add a print that says something like **Confusion matrix for adversarial examples generated and evaluated on a CNN with 2 layers** and then for a transferability confusion matrix **Confusion matrix for adversarial examples generated on a CNN with 2 layers and evaluated on a CNN with 4 layers**

The number of layers are not exact but it should just indicate to the user what is the high-level difference between the two architectures.

**Comment by npapernot** [comment]
Thank you for the pull request! I left a few comments in my review above. Feel free to ping me if you have any questions

**Comment by goodfeli** [comment]
I think it's important to review this properly, but it arrives at a time of very high reviewer load (due to preparation for the 2.0 release). I propose leaving this PR in place until the preceding PRs have been dealt with.

**Comment by npapernot** [comment]
Agreed

**Comment by goodfeli** [comment]
Now that the model abstraction is merged, it's OK to revisit this. I see there are changes requested by @npapernot .

**Comment by abhibhav14** [comment]
I will make those changes and also change the tutorial to use the new model abstraction.

**Comment by npapernot** [comment]
Just so you know there will also be changes pushed to the `generate_np` method per what was discussed on the mailing list. The new `generate_np` method will be merged first (because #156) has been rebased multiple times already).

**Comment by npapernot** [comment]
@abhibhav14  are you still willing to work on this PR?

**Comment by abhibhav14** [comment]
@npapernot  Given the extent of changes to the library, I think it would be best to close this PR and create a new one.

**Comment by npapernot** [comment]
I agree, if that is easier for you it would be best to open a new PR

**Pull request closed without merging**